cmd/snowflake-exporter/main.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 Grafana Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/snowflake-prometheus-exporter/collector"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/promlog"
+	"github.com/prometheus/common/promlog/flag"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/exporter-toolkit/web"
+	webflag "github.com/prometheus/exporter-toolkit/web/kingpinflag"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	webConfig  = webflag.AddFlags(kingpin.CommandLine, ":9975")
+	metricPath = kingpin.Flag("web.telemetry-path", "Path under which to expose metrics.").Default("/metrics").Envar("SNOWFLAKE_EXPORTER_WEB_TELEMETRY_PATH").String()
+	account    = kingpin.Flag("account", "The account to collect metrics for.").Envar("SNOWFLAKE_EXPORTER_ACCOUNT").Required().String()
+	username   = kingpin.Flag("username", "The username for the user used when querying metrics.").Envar("SNOWFLAKE_EXPORTER_USERNAME").Required().String()
+	password   = kingpin.Flag("password", "The password for the user used when querying metrics.").Envar("SNOWFLAKE_EXPORTER_PASSWORD").Required().String()
+	role       = kingpin.Flag("role", "The role to use when querying metrics.").Default("ACCOUNTADMIN").Envar("SNOWFLAKE_EXPORTER_ROLE").String()
+	warehouse  = kingpin.Flag("warehouse", "The warehouse to use when querying metrics. If none is specified, the default warehouse for the user will be used.").Envar("SNOWFLAKE_EXPORTER_WAREHOUSE").String()
```

Comment:
The mention of using a default warehouse doesn't appear to work if this parameter is not specified, at least with a brand new Snowflake account that only has one warehouse:
```bash
ERRO[0020]connection.go:318 gosnowflake.(*snowflakeConn).queryContextInternal error: 000606 (57P03): No active warehouse selected in the current session.  Select an active warehouse with the 'use warehouse' command. 
ts=2022-12-12T10:40:12.852Z caller=collector.go:320 level=error msg="Failed to collect replication metrics." err="failed to query metrics: 000606 (57P03): No active warehouse selected in the current session.  Select an active warehouse with the 'use warehouse' command.\n"```

Reply:
Yeah, the idea was that you'd have to set the default warehouse for the user using the `ALTER USER` command (https://docs.snowflake.com/en/sql-reference/sql/alter-user.html) before hand.

I think we might be better off just requiring that a warehouse is specified, to avoid leaving it out when there is no default.

Reply:
Makes sense, that sounds like a reasonable solution to me! :)